azure-quantum/tests/quantum_test_base.py

```diff
@@ -0,0 +1,197 @@
+
+        regex_replacer = CustomRecordingProcessor()
+        recording_processors = []
+        recording_processors.append(regex_replacer)
+        recording_processors.append(AccessTokenReplacer())
+        replay_processors = []
+        
+        super(QuantumTestBase, self).__init__(method_name, recording_processors=recording_processors, replay_processors=replay_processors)
+
+        if not (self.in_recording or self.is_live):
+            self._client_id = self.dummy_uid
+            self._client_secret = self.dummy_clientsecret
+            self._tenant_id = self.dummy_uid
+            self._resource_group = self.dummy_rg
+            self._subscription_id = self.dummy_uid
+            self._workspace_name = self.dummy_ws
+
+        regex_replacer.register_regex(self.client_id, self.dummy_uid)
+        regex_replacer.register_regex(self.client_secret, self.dummy_clientsecret)
+        regex_replacer.register_regex(self.tenant_id, self.dummy_uid)
+        regex_replacer.register_regex(self.subscription_id, self.dummy_uid)
+        regex_replacer.register_regex(self.workspace_name, self.dummy_ws)
+        regex_replacer.register_regex(self.resource_group, self.dummy_rg)
+        regex_replacer.register_regex(r'/subscriptions/([a-f0-9]+[-]){4}[a-f0-9]+', "/subscriptions/" + self.dummy_uid)
+        regex_replacer.register_regex(r'job-([a-f0-9]+[-]){4}[a-f0-9]+', "job-" + self.dummy_uid)
+        regex_replacer.register_regex(r'jobs/([a-f0-9]+[-]){4}[a-f0-9]+', "jobs/" + self.dummy_uid)
+        regex_replacer.register_regex(r'"id":\s*"([a-f0-9]+[-]){4}[a-f0-9]+"', '"id": "{}"'. format(self.dummy_uid))
+        regex_replacer.register_regex(r'/resourceGroups/[a-z0-9-]+/', "/resourceGroups/dummy-rg/")
+        regex_replacer.register_regex(r'/workspaces/[a-z0-9-]+/', "/workspaces/dummy-ws/")
+        regex_replacer.register_regex(r'sig=[0-9a-zA-Z%]+\&', "sig=sanitized&")
+        regex_replacer.register_regex(r'sv=[^&]+\&', "sv=sanitized&")
+        regex_replacer.register_regex(r'se=[^&]+\&', "se=sanitized&")
+
+    def setUp(self):
+        super(QuantumTestBase, self).setUp()
+        # mitigation for issue https://github.com/kevin1024/vcrpy/issues/533
+        self.cassette.allow_playback_repeats = True
+
+    @property
+    def client_id(self):
+        return self._client_id
+
+    @property
+    def client_secret(self):
+        return self._client_secret
+
+    @property
+    def tenant_id(self):
+        return self._tenant_id
+
+    @property
+    def resource_group(self):
+        return self._resource_group
+
+    @property
+    def subscription_id(self):
+        return self._subscription_id
+
+    @property
+    def workspace_name(self):
+        return self._workspace_name
+
+    def create_workspace(self) -> Workspace:
+        """Create workspace using credentials stored in config file
+
+        :return: Workspace
+        :rtype: Workspace
+        """
```

Comment:
```suggestion
    def create_workspace(self, resource_url: str = "https://quantum.microsoft.com") -> Workspace:
        """Create workspace using credentials stored in config file

        :param resource_url: Resource URL to use
        :type resource_url: str
        :return: Workspace
        :rtype: Workspace
        """
```

Reply:
This code has been refactored and we don't have use the `resource url` anymore. All applicable PEP8 fixes were applied with #53, #54, #56.